src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -34,9 +41,27 @@ public TestCase persistTestCase(TestCase testCase, String measureId) {
     return testCase;
   }
 
-  public TestCase updateTestCase(TestCase testCase, String measureId) {
+  public TestCase updateTestCase(TestCase testCase, String measureId, String username) {
     Measure measure = findMeasureById(measureId);
+    List<TestCase> tcs = measure.getTestCases();
+    Instant now = Instant.now();
+    testCase.setLastModifiedAt(now);
+    testCase.setLastModifiedBy(username);
+    int idx =
+        IntStream.range(0, tcs.size())
+            .filter(i -> tcs.get(i).getId().equals(testCase.getId()))
+            .findFirst()
+            .orElse(-1);
+    if (idx > -1) {
+      TestCase existing = tcs.get(idx);
+      testCase.setCreatedAt(existing.getCreatedAt());
+      testCase.setCreatedBy(existing.getCreatedBy());
```

Comment:
```suggestion
      testCase.setCreatedBy(existing.getCreatedBy());
      measure.getTestCases().remove(existing);
```

Haven't tried it out. Since we'll have the existing test case, do you think it makes sense to remove it here instead of hunting it up again and removing it on line 63?

Reply:
ah yes, good idea - in fact, remove(index) returns the object so the logic can be reduced down to a single call in the first if block

Reply:
updated